ql.go

```diff
@@ -516,6 +528,33 @@ func (r *whereRset) planUnaryOp(x *unaryOperation) (plan, error) {
 }
 
 func (r *whereRset) plan(ctx *execCtx) (plan, error) {
+	if r.sel != nil {
+		var exists bool
+		p, err := r.sel.plan(ctx)
+		if err != nil {
+			return nil, err
+		}
+		err = p.do(ctx, func(i interface{}, data []interface{}) (bool, error) {
+			if len(data) > 0 {
```

Comment:
> Is it possible that the first result set might be zero but the second might not be zero?

I guess you mean rows, not result sets.

There's only one result set. Result set is a set of rows. The set is empty if it consists of zero rows and non empty if it consists of at least one row.

The `EXISTS ( SELECT ...)` evaluates to true if the parenthesized select's result set is non empty, so seeing any row (eg. the first) is enough.

The test `len(data) > 0` checks if a row of a result set has non zero number of columns. All rows of a result set have the same number of columns and so far the number of columns can never be zero. (`SELECT;` is invalid as is `SELECT FROM t;`).